public/API/API_GetConsoleIDs.php

```diff
@@ -4,20 +4,36 @@
 
 /*
  *  API_GetConsoleIDs - returns mapping of known consoles
+ *    a : active - 1 for only active systems, 0 for all (default: 0)
+ *    g : only game systems - 1 for only game systems, 0 for all system types (Events, Hubs, etc) (default: 0)
  *
  *  array
  *   object    [value]
- *    string    ID                  unique identifier of the console
+ *    int       ID                  unique identifier of the console
  *    string    Name                name of the console
  *    string    IconURL             system icon URL
+ *    bool      Active              boolean value indicating if the console is active in RA
+ *    bool      IsGameSystem        boolean value indicating if is a game system (not Events, Hubs, etc.)
  */
 
-$systems = System::all()->map(function ($system) {
-    return [
+$onlyActive = (bool) request()->query('a', '0');
+$onlyGameConsoles = (bool) request()->query('g', '0');
+
+$systems = System::all()
+    ->filter(function ($system) use ($onlyActive) {
+        return $onlyActive ? isValidConsoleId($system->ID) : true;
+    })
+    ->filter(function ($system) use ($onlyGameConsoles) {
+        return $onlyGameConsoles ? System::isGameSystem($system->ID) : true;
+    });
```

Comment:
It's more efficient to do this filtering at the query level.

Reply:
Since there is no database fields to query around, how it would be? Sorry, don't have much experience with Laravel.

Reply:
Ah, I was mistaken. You're right. Apologies.